### DISCUSSION TITLE
History pane equivalent or work around?

### DISCUSSION THREAD

**Discussion by mark-andrews** [posit-dev/positron#5480] [Q&A] [answered]
Two things that I miss from RStudio that appear to be not present in Positron are the Packages pane and the History pane. I have seen discussions relating to the packages pane functionality, e.g. https://github.com/posit-dev/positron/discussions/4409 https://github.com/posit-dev/positron/issues/1369, but I don't think there has been a discussion about the RStudio history pane functionality.

What I find the RStudio history pane most useful for is the "To source" button. Often, I play around in the console. My commands get recorded in history. Eventually, after I have got things working, I move these commands into a script. This is made very easy by browsing though my history in the history pane and selecting and clicking "To source". In Positron, after I play around in the console, I have to do ctrl+R to search through history (which I can also do in RStudio; very handy) and then bring back up commands and then copy and paste. It feels a bit clunky compared to what can be accomplished in the history pane.

Are there any thoughts on whether an equivalent of the history pane will be in Positron, or is there some other way that I am not aware of to easily move commands from history into the files being edited?

**Comment by petetronic** [comment, marked as answer]
Thank you @mark-andrews for helping testing Positron. I believe Ctrl+R and up/down navigation is currently the main workflow available. You can highlight a historical command, right-click or ctrl-click to get a Copy context menu, and then paste it into source, but it's a little cumbersome. I've logged this as  https://github.com/posit-dev/positron/issues/5484 to capture the enhancement request asking for equivalent functionality as RStudio's History pane.
